### PR TITLE
Collected small changes and fixes

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This pull request contains multiple small changes and fixes that do not warrant a pull request of their own.

**Related Issue(s)**

Fixes the bug reported here: https://matsci.org/t/bug-in-angle-cosine-periodic-from-extra-molecule/52963
Fixes the issue reported here: https://github.com/lammps/lammps/pull/4037#issuecomment-1889939953

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Implementation Notes**

The following individual changes are included:
- bugfix for angle style cosine/periodic with multiplicity 1
- update unit test for angle style cosine/periodic to include multiplicity 1 in angle_coeff settings
- BPM package must set a define so that `delete_bonds` only accesses fix bond/history data when BPM is included
- similar changes are required for fix `bond/break`, `fix bond/swap`, and `fix bond/react`
- fix bug in fix update/special/bonds
- fix bug in pair style slater/coul/long/gpu (from @ndtrung81)
- fix typo in OpenMP neighbor lists class for r-RESPA
- improved error messages
- remove a few obsolete machine makefiles
- whitespace fixes

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

### CODE REVIEW

File: src/delete_bonds.cpp
Comment (jtclemm) [inline src/delete_bonds.cpp:1]: Wow, totally overlooked this. Similar changes will be needed to various fixes in the MC package...

Comment (akohlmey) [reply src/delete_bonds.cpp:1]: ... and to fix bond/react.

**Review by jrgissing — APPROVED** [review]

File: src/delete_bonds.cpp
Comment (jtclemm) [reply src/delete_bonds.cpp:1]: Thank you for fixing those errors.

**Review by jtclemm — APPROVED** [review]

File: src/delete_bonds.cpp
Comment (akohlmey) [reply src/delete_bonds.cpp:1]: No worries. This is what I do and these kind of straightforward fixes are welcome after struggling with a bunch or hard to pin down issue.
Also, it is not just your fault. This issue should have been detected by integration testing, but only the GPU unit tests (which are only run *after* a merge or when requested explicitly) did not include the BPM package and thus failed.

**Review by sjplimp — APPROVED** [review]

**Comment by sjplimp** [comment]
@akohlmey Do you like the solution of adding #ifdefs to the MC fixes better than just moving fix_bond_history.cpp back into src, rather than the BPM package, where it was before ?

**Comment by akohlmey** [comment]
> @akohlmey Do you like the solution of adding #ifdefs to the MC fixes better than just moving fix_bond_history.cpp back into src, rather than the BPM package, where it was before ?

I am a bit conflicted about this one. Moving the fix back would make things simpler, but the ifdefs indicate that this is a feature related to BPM (only). That is why I was waiting for your review since there was a possibility that you would object.

**Comment by akohlmey** [comment]
@jtclemm please have a look at commits e3f81a25130d80fcf436f25be3d46519ec2809ce e82ba61183d7f6b8faa63fc52cfc04675fe4c1bd and let me know if I have guessed the correct changes. These were spotted through checking out compiler warnings issued by Clang 17 and static code analysis reports by Coverity Scan (https://scan.coverity.com/).

**Comment by sjplimp** [comment]
@akohlmey I think moving fix_bond_history back into src is probably best, treating it as a utility class which can be used by multiple commands, similar to fix_neigh_history.  Joel and I thought it was only used by BPM but the delete_bonds command is another user.  Basically any model that wants to store time history values for a pairwise or bond interaction could use these 2 classes.

**Comment by akohlmey** [comment]
> I think moving fix_bond_history back into src is probably best

Like this?

**Comment by sjplimp** [comment]
yes - thanks

**Comment by jtclemm** [comment]
> @jtclemm please have a look at commits [e3f81a2](https://github.com/lammps/lammps/commit/e3f81a25130d80fcf436f25be3d46519ec2809ce) [e82ba61](https://github.com/lammps/lammps/commit/e82ba61183d7f6b8faa63fc52cfc04675fe4c1bd) and let me know if I have guessed the correct changes. These were spotted through checking out compiler warnings issued by Clang 17 and static code analysis reports by Coverity Scan

These changes look correct. Thanks again for fixing the mistakes!

**Pull request merged**
